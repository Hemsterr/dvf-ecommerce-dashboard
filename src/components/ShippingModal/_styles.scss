.shipping-modal {
  text-align: center;

  &__title {
    font-family: $font-base-futura;
    color: $black;
    @include text-size(pxToRem(12px), pxToRem(1px), pxToRem(30px));
    padding-bottom: pxToRem(8px);
  }

  &__address {
    margin: 0 auto;
  }

  &__stateWrapper {
    display: flex;
  }

  &__input {
    justify-content: center;
    padding-bottom: pxToRem(10px);
    & input {
      width: pxToRem(270px);
    }
  }

  & .input-group__label {
    width: pxToRem(85px);
    text-align: right;
    font-family: $font-base-futura;
  }

  &__state {
    width: pxToRem(200px);
    flex-basis: 50%;
    justify-content: flex-end;

    & input {
      width: pxToRem(55px);
    }
  }

  &__zipCode {
    width: pxToRem(200px);
    flex-basis: 50%;
    padding: 0;
    justify-content: flex-start;

    & input {
      width: pxToRem(90px);
    }

    & .input-group__label {
      width: pxToRem(63px);
    }
  }

  &__btn {
    width: pxToRem(200px);
    margin-top: pxToRem(60px);
  }
}
