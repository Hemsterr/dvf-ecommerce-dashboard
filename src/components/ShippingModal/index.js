// @flow
// Libs
import React, { useState } from 'react'

// Components
import Modal from '../Modal'
import Input from '../Input'
import Button from '../Button'

// Constant
import { shippingAddress } from '../../constants/alterations'
import { shippingAddressValidation } from '../../helpers/alterations'

type Props = {
  handleCloseModal: Function,
}

const ShippingModal = (props: Props) => {
  const { handleCloseModal } = props
  const [user, setUser] = useState({})
  const [error, setError] = useState({})
  const handleSetUserData = (value, key) => {
    user[key] = value
    setUser(user)
  }

  const confirmShippingAddress = () => {
    const errorMessage = shippingAddressValidation(user)
    if (Object.keys(errorMessage).length) {
      setError(errorMessage)
    } else {
    }
  }
  console.log(error)
  return (
    <Modal className="shipping-modal" handleCloseModal={handleCloseModal}>
      <div className="shipping-modal">
        <p className="shipping-modal__title">Request a Shipping Label</p>
        <p className="shipping-modal__title">
          Please confirm email & shipping address
        </p>
        <div className="shipping-modal__address">
          {shippingAddress.map(item => (
            <Input
              key={item.key}
              label={item.label}
              className="shipping-modal__input"
              isBorder
              handleOnBlur={event => handleSetUserData(event, item.key)}
              errorMessage={error[item.key]}
            />
          ))}
          <div className="shipping-modal__stateWrapper">
            <Input
              label="State"
              className="shipping-modal__input shipping-modal__state col-xs-12 col-sm-12"
              isBorder
              handleOnBlur={event => handleSetUserData(event, 'state')}
              errorMessage={error.state}
            />
            <Input
              label="Zip Code"
              className="shipping-modal__input shipping-modal__zipCode col-xs-12 col-sm-12"
              isBorder
              handleOnBlur={event => handleSetUserData(event, 'zipCode')}
              errorMessage={error.zipCode}
            />
          </div>
          <Button
            className="btn btn__basic shipping-modal__btn"
            label="Confirm"
            type="success"
            size="small"
            handleOnClick={() => confirmShippingAddress()}
          />
        </div>
      </div>
    </Modal>
  )
}

ShippingModal.defaultProps = {
  error: {},
  handleCloseModal: () => {},
}

export default ShippingModal
