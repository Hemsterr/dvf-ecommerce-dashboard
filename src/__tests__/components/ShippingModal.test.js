// Libs
import React from 'react'
import { render, fireEvent } from '@testing-library/react'

// Component
import ShippingModal from '../../components/ShippingModal'
import ERROR_MESSAGES from '../../constants/errorMessages'

const props = {
  handleCloseModal: jest.fn(),
  children: '',
  title: 'Appointments detail',
  className: 'isUpcoming',
}

const wrapper = (props = {}) => {
  const component = render(<ShippingModal {...props} />)

  return component
}

describe('ShippingModal component', () => {
  // Snapshot
  test('should render correctly ShippingModal component default props', () => {
    const { container } = wrapper(props)
    expect(container).toMatchSnapshot()
    expect(ShippingModal.defaultProps.handleCloseModal()).toBeUndefined()
  })

  test('Testing event onBlur textinput', () => {
    const { container, rerender } = wrapper(props)
    const formInput = container.querySelectorAll('input')
    expect(formInput.length).toEqual(7)

    const button = container.querySelector('.btn')
    fireEvent.click(button)
    rerender(<ShippingModal {...props} />)
    const messageError = container.querySelectorAll('.input-group__label-error')
    expect(messageError.length).toEqual(6)

    // Testing input invalid email
    fireEvent.blur(formInput[0], {
      target: { value: '0123456' },
    })
    fireEvent.click(button)
    expect(messageError[0].textContent).toEqual(ERROR_MESSAGES.EMAIL_FORMAT)

    // Testing input invalid zip code
    fireEvent.blur(formInput[6], {
      target: { value: 'zip code' },
    })
    fireEvent.click(button)
    expect(messageError[5].textContent).toEqual(
      ERROR_MESSAGES.INVALID_ZIPCODE_FORMAT
    )

    // Testing input valid value
    fireEvent.blur(formInput[0], {
      target: { value: 'join@gmai.com' },
    })

    fireEvent.blur(formInput[1], {
      target: { value: 'join' },
    })

    fireEvent.blur(formInput[2], {
      target: { value: '123 st' },
    })

    fireEvent.blur(formInput[4], {
      target: { value: 'New York' },
    })

    fireEvent.blur(formInput[5], {
      target: { value: 'NY' },
    })

    fireEvent.blur(formInput[6], {
      target: { value: '12345' },
    })

    fireEvent.click(button)
    rerender(<ShippingModal {...props} />)
    expect(
      container.querySelectorAll('.input-group__label-error').length
    ).toEqual(0)

    // Testing render contact modal
    expect(
      container.querySelectorAll('.shipping-modal__header').length
    ).toEqual(1)
  })
})
