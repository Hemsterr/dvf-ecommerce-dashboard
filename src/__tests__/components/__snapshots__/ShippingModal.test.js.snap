// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingModal component should render correctly ShippingModal component default props 1`] = `
<div>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal shipping-modal"
    >
      <div
        class="modal__dialog"
      >
        <div
          class="modal__header"
          role="presentation"
        >
          <div
            class="modal__icon"
          />
        </div>
        <p
          class="modal__title"
        />
        <div
          class="modal__body"
        >
          <div
            class="shipping-modal"
          >
            <p
              class="shipping-modal__title"
            >
              Request a Shipping Label
            </p>
            <p
              class="shipping-modal__title"
            >
              Please confirm email & shipping address
            </p>
            <div
              class="shipping-modal__address"
            >
              <div
                class="shipping-modal__input input-group"
              >
                <label
                  class="input-group__label"
                >
                  Email
                </label>
                <div>
                  <input
                    class="input-group__text  input-group__border"
                    placeholder=""
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <div
                class="shipping-modal__input input-group"
              >
                <label
                  class="input-group__label"
                >
                  Name
                </label>
                <div>
                  <input
                    class="input-group__text  input-group__border"
                    placeholder=""
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <div
                class="shipping-modal__input input-group"
              >
                <label
                  class="input-group__label"
                >
                  Address 1
                </label>
                <div>
                  <input
                    class="input-group__text  input-group__border"
                    placeholder=""
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <div
                class="shipping-modal__input input-group"
              >
                <label
                  class="input-group__label"
                >
                  Address 2
                </label>
                <div>
                  <input
                    class="input-group__text  input-group__border"
                    placeholder=""
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <div
                class="shipping-modal__input input-group"
              >
                <label
                  class="input-group__label"
                >
                  City
                </label>
                <div>
                  <input
                    class="input-group__text  input-group__border"
                    placeholder=""
                    type="text"
                    value=""
                  />
                  
                </div>
              </div>
              <div
                class="shipping-modal__stateWrapper"
              >
                <div
                  class="shipping-modal__input shipping-modal__state col-xs-12 col-sm-12 input-group"
                >
                  <label
                    class="input-group__label"
                  >
                    State
                  </label>
                  <div>
                    <input
                      class="input-group__text  input-group__border"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    
                  </div>
                </div>
                <div
                  class="shipping-modal__input shipping-modal__zipCode col-xs-12 col-sm-12 input-group"
                >
                  <label
                    class="input-group__label"
                  >
                    Zip Code
                  </label>
                  <div>
                    <input
                      class="input-group__text  input-group__border"
                      placeholder=""
                      type="text"
                      value=""
                    />
                    
                  </div>
                </div>
              </div>
              <button
                class="btn btn__basic shipping-modal__btn btn__small btn__success"
                type="button"
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal__backdrop"
      role="presentation"
    />
  </div>
</div>
`;
